presto-orc/src/main/java/com/facebook/presto/orc/ByteArrayUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+
+import static io.airlift.slice.SizeOf.SIZE_OF_LONG;
+import static java.lang.Math.min;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+
+public class ByteArrayUtils
+{
+    private static final Unsafe unsafe;
+
+    static {
+        try {
+            // fetch theUnsafe object
+            Field field = Unsafe.class.getDeclaredField("theUnsafe");
+            field.setAccessible(true);
+            unsafe = (Unsafe) field.get(null);
+            if (unsafe == null) {
+                throw new RuntimeException("Unsafe access not available");
+            }
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private ByteArrayUtils() {}
+
+    public static int compareRanges(byte[] left, int leftOffset, int leftLength, byte[] right, int rightOffset, int rightLength)
+    {
+        int minLength = min(leftLength, rightLength);
+        long i = 0;
+        if (minLength >= SIZE_OF_LONG) {
+            while (i < minLength) {
+                long leftLong = unsafe.getLong(left, ARRAY_BYTE_BASE_OFFSET + leftOffset + i);
+                long rightLong = unsafe.getLong(right, ARRAY_BYTE_BASE_OFFSET + rightOffset + i);
+
+                if (leftLong != rightLong) {
+                    return longBytesToLong(leftLong) < longBytesToLong(rightLong) ? -1 : 1;
+                }
+
+                i += SIZE_OF_LONG;
+            }
+            i -= SIZE_OF_LONG;
+        }
+
+        while (i < minLength) {
+            int leftInt = unsignedByteToInt(unsafe.getByte(left, ARRAY_BYTE_BASE_OFFSET + leftOffset + i));
```

Comment:
nit: Should this be called `leftByte` and `rightByte` ? -- Because when I read through the code, I am thinking "hmm, before we are comparing long by long (8 bytes), and now we are going to comparing integer by integer (4 bytes)"